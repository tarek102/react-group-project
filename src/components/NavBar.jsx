import React from 'react';
import { Navbar, Nav, Container } from 'react-bootstrap';
import { LinkContainer } from 'react-router-bootstrap';
import planet from './planet.png';

function NavBar() {
  return (
    <Navbar activekey={window.location.pathname} expand="md" sticky="top" className="Navbar bg-white">
      <Container className="NavContainer">
        <LinkContainer to="/space_travelers_hub/" rel="noopener noreferrer">
          <Navbar.Brand>
            <img src={planet} alt="logo" />
            <h3>Space Travelers&apos; Hub</h3>
          </Navbar.Brand>
        </LinkContainer>
        <Navbar.Toggle aria-controls="responsive-navbar-nav" />
        <Navbar.Collapse>
          <Nav className="ms-auto d-flex">
            <LinkContainer to="/" rel="noopener noreferrer">
              <Nav.Link>Rockets</Nav.Link>
            </LinkContainer>

            <LinkContainer to="/Missions" rel="noopener noreferrer">
              <Nav.Link>Missions</Nav.Link>
            </LinkContainer>

            <LinkContainer to="/MyProfile/" rel="noopener noreferrer">
              <Nav.Link>
                <span>|</span>
                <span>My Profile</span>
              </Nav.Link>
            </LinkContainer>
          </Nav>
        </Navbar.Collapse>
      </Container>

    </Navbar>

  );
}

export default NavBar;
