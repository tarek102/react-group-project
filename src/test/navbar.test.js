import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import NavBar from '../components/NavBar';

it('The Navbar works properly', () => {
  const nav = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <NavBar />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(nav).toMatchSnapshot();
});
