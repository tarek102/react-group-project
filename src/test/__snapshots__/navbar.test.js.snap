// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Navbar works properly 1`] = `
<nav
  activekey="/"
  className="Navbar bg-light navbar navbar-expand-md navbar-light sticky-top"
>
  <div
    className="NavContainer container"
  >
    <a
      className="navbar-brand"
      href="/space_travelers_hub/"
      onClick={[Function]}
      rel="noopener noreferrer"
      style={null}
    >
      Space Travelers' Hub
    </a>
    <button
      aria-controls="responsive-navbar-nav"
      aria-label="Toggle navigation"
      className="navbar-toggler collapsed"
      onClick={[Function]}
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      aria-expanded={null}
      className="navbar-collapse collapse"
    >
      <div
        className="ms-auto d-flex navbar-nav"
        onKeyDown={[Function]}
      >
        <a
          className="active nav-link"
          data-rr-ui-event-key="/"
          disabled={false}
          href="/"
          onClick={[Function]}
          rel="noopener noreferrer"
          style={Object {}}
        >
          Rockets
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="/Missions"
          disabled={false}
          href="/Missions"
          onClick={[Function]}
          rel="noopener noreferrer"
          style={null}
        >
          Missions
        </a>
        <a
          className="nav-link"
          data-rr-ui-event-key="/MyProfile/"
          disabled={false}
          href="/MyProfile/"
          onClick={[Function]}
          rel="noopener noreferrer"
          style={null}
        >
          <span>
            |
          </span>
          <span>
            My Profile
          </span>
        </a>
      </div>
    </div>
  </div>
</nav>
`;
